homeassistant/components/media_player/universal.py

```diff
@@ -48,113 +51,68 @@
-    for key in del_keys:
-        del config[key]
-
-    return True
-
-
-def validate_children(config):
-    """Validate children."""
-    if CONF_CHILDREN not in config:
-        _LOGGER.info(
-            "No children under Universal Media Player (%s)", config[CONF_NAME])
-        config[CONF_CHILDREN] = []
-    elif not isinstance(config[CONF_CHILDREN], list):
-        _LOGGER.warning(
-            "Universal Media Player (%s) children not list in config. "
-            "They will be ignored", config[CONF_NAME])
-        config[CONF_CHILDREN] = []
-
-
-def validate_commands(config):
-    """Validate commands."""
-    if CONF_COMMANDS not in config:
-        config[CONF_COMMANDS] = {}
-    elif not isinstance(config[CONF_COMMANDS], dict):
-        _LOGGER.warning(
-            "Universal Media Player (%s) specified commands not dict in "
-            "config. They will be ignored", config[CONF_NAME])
-        config[CONF_COMMANDS] = {}
-
-
-def validate_attributes(config):
-    """Validate attributes."""
-    if CONF_ATTRS not in config:
-        config[CONF_ATTRS] = {}
-    elif not isinstance(config[CONF_ATTRS], dict):
-        _LOGGER.warning(
-            "Universal Media Player (%s) specified attributes "
-            "not dict in config. They will be ignored", config[CONF_NAME])
-        config[CONF_ATTRS] = {}
-
-    for key, val in config[CONF_ATTRS].items():
-        attr = val.split('|', 1)
-        if len(attr) == 1:
-            attr.append(None)
-        config[CONF_ATTRS][key] = attr
-
-
 class UniversalMediaPlayer(MediaPlayerDevice):
     """Representation of an universal media player."""
 
-    def __init__(self, hass, name, children, commands, attributes):
+    def __init__(self, hass, name, children,
+                 commands, attributes, state_template=None):
         """Initialize the Universal media device."""
         self.hass = hass
         self._name = name
         self._children = children
         self._cmds = commands
-        self._attrs = attributes
+        self._attrs = {}
+        for key, val in attributes.items():
+            attr = val.split('|', 1)
+            if len(attr) == 1:
+                attr.append(None)
+            self._attrs[key] = attr
         self._child_state = None
+        self._state_template = state_template
 
         @callback
         def async_on_dependency_update(*_):
             """Update ha state when dependencies update."""
             self.async_schedule_update_ha_state(True)
 
         depend = copy(children)
-        for entity in attributes.values():
+        for entity in self._attrs.values():
             depend.append(entity[0])
+        if self._state_template is not None:
+            self._state_template.hass = hass
+            for entity in self._state_template.extract_entities():
+                depend.append(entity)
 
-        async_track_state_change(hass, depend, async_on_dependency_update)
+        async_track_state_change(hass, list(set(depend)),
```

Comment:
please move that into `async_added_to_hass` and use new style: `hass.helpers.event.async_track_state_change()`

Reply:
You mean calling it like: 
`self.hass.helpers.event.async_track_state_change(self.hass, list(set(depend)), async_on_dependency_update)` instead of importing the method?

I'm on it